src/osd/ReplicatedPG.cc

```diff
@@ -8303,6 +8301,7 @@ void ReplicatedPG::eval_repop(RepGather *repop)
 	 repop->on_committed.erase(p++)) {
       (*p)();
     }
+    publish_stats_to_osd();
```

Comment:
Uh, maybe? This is where it was located previously, and immediately follows where it would be called as part of an on_complete lambda, so I put it back as that's the minimal change and I'm unsure if there are any side effects to be worried about. (Though I definitely don't see anything that makes me think publish_stats_to_osd() is sensitive to much of anything.)